accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/AccessioningService.java

```diff
@@ -35,23 +35,36 @@
     List<AccessionWrapper<MODEL, HASH, ACCESSION>> get(List<? extends MODEL> accessionedObjects);
 
     /**
-     * Finds last version of all valid accessions with their possible data model representations.
+     * Finds last version of provided accession with its possible data model representations.
      *
-     * @param accessions
-     * @return
+     * @param accession
+     * @return latest version of provided accession
+     * @throws AccessionDoesNotExistException when either accession has never existed.
+     * @throws AccessionDeprecatedException   when either accession exists but has been deprecated
      */
-    List<AccessionWrapper<MODEL, HASH, ACCESSION>> getByAccessions(List<ACCESSION> accessions);
+    AccessionWrapper<MODEL, HASH, ACCESSION> getByAccession(ACCESSION accession)
+            throws AccessionDoesNotExistException, AccessionDeprecatedException;
 
-    AccessionWrapper<MODEL, HASH, ACCESSION> getByAccessionAndVersion(ACCESSION accessions, int version)
-    throws AccessionDoesNotExistException, AccessionMergedException, AccessionDeprecatedException;
+    /**
+     * Finds last version of provided accession with its possible data model representations.
+     *
+     * @param accession
+     * @param version
+     * @return accession with particular version
+     * @throws AccessionDoesNotExistException when either accession has never existed.
+     * @throws AccessionDeprecatedException   when either accession exists but has been deprecated
+     * @throws AccessionMergedException       when the accession exists but has been merged into another accession
+     */
+    AccessionWrapper<MODEL, HASH, ACCESSION> getByAccessionAndVersion(ACCESSION accession, int version)
+            throws AccessionDoesNotExistException, AccessionMergedException, AccessionDeprecatedException;
 
     /**
      * Updates a specific patch version of an accessioned object. It does not create a new version / patch
      *
      * @param accession
      * @param version
      * @param message
-     * @return Accession with complete patch information
+     * @return Accession with updated information
```

Comment:
The response of this method is the updated Accession with all the patch information.